### PR TITLE
fix(animation): make Simple1D BlendTree duration/eval agree for unbound param (#410)

### LINKED ISSUE #410: BlendTree Simple1D: GetDuration() and Evaluate() disagree when BlendParameterX is unbound
## Summary

For a **`Simple1D` blend tree with no blend parameter bound** (empty `BlendParameterX`), `BlendTree::GetDuration()` and `BlendTree::Evaluate()` disagree on how to treat the unconfigured tree, so the tree can play its pose at the wrong speed.

This is a **pre-existing latent inconsistency** (present on `master`), surfaced during review of #406 but deferred — fixing it changes runtime playback semantics, which is out of scope for that mechanical SonarQube cleanup. CodeRabbit and the reviewer agreed to track it separately ([#406 thread](https://github.com/drsnuggles8/OloEngineBase/pull/406#discussion_r3483145838)).

## Detail

In [`OloEngine/src/OloEngine/Animation/BlendTree.cpp`](OloEngine/src/OloEngine/Animation/BlendTree.cpp):

- **`Evaluate()`** (Simple1D case) always samples the pose from `params.GetFloat(BlendParameterX)`. With an empty `BlendParameterX`, `AnimationParameterSet::GetFloat("")` returns `0.0f`, so `Evaluate1D(0.0f, ...)` samples the **lowest-threshold child** region.
- **`GetDuration()`** (Simple1D case) special-cases the empty param to `ComputeAverageDuration(Children)` — the **plain average of all children's** durations.

So an unconfigured 1D tree normalizes/advances time using the *average* duration of all children, but renders the pose of (roughly) the *first* child. If the first child's natural duration differs from the average, the animation plays at the wrong speed.

### Concrete example
Three children with durations `1s, 2s, 3s`, no `BlendParameterX` bound:
- `GetDuration()` → `2s` (average)
- `Evaluate()` → samples the `1s` child's pose

→ the 1s clip is timed as if it were 2s, i.e. plays at half speed.

## Suggested fix (either)

1. **Make `GetDuration()` share `Evaluate()`'s fallback** — drop the empty-param special case so both use `params.GetFloat(BlendParameterX)` (i.e. `Compute1DDuration(Children, params.GetFloat(BlendParameterX))`). Because both call `GetFloat` with the same (empty) key, they're guaranteed consistent. This also removes the now-unused `ComputeAverageDuration` helper (no other callers). This was effectively the behavior before the #406 refactor's flow was made explicit.
2. **Reject/validate empty `BlendParameterX` for `Simple1D`** at authoring/load time, so an unconfigured tree can't run with mismatched duration vs. pose sampling.

Option 1 is the smaller, lower-risk change; option 2 is stricter but needs validation plumbing and may affect existing scenes.

## Notes
- Whichever path is chosen, add a `BlendTree` duration/eval-consistency unit test for the empty-`BlendParameterX` `Simple1D` case (the current `BlendTree`/animation tests don't cover it).
- Pre-existing; **not** introduced by #406.

### PR DESCRIPTION
**Pull request by drsnuggles8**
## Summary

Fixes #410 — `BlendTree::GetDuration()` and `BlendTree::Evaluate()` disagreed for a `Simple1D` tree with **no blend parameter bound** (empty `BlendParameterX`).

- `Evaluate()` always sampled `params.GetFloat(BlendParameterX)`, which returns `0.0f` for an empty key → it rendered the **lowest-threshold child**.
- `GetDuration()` special-cased the empty param to `ComputeAverageDuration()` — a plain average of *all* children's durations.

So an unconfigured 1D tree advanced time using the **average** duration while rendering (roughly) the **first** child's pose → it played at the wrong speed. Pre-existing latent bug; surfaced during the #406 review and deferred there as out of scope.

## Fix (issue's "Option 1" — smallest, lowest-risk)

- `GetDuration()`'s Simple1D path is now just `Compute1DDuration(Children, params.GetFloat(BlendParameterX))`. Because both `GetDuration()` and `Evaluate()` now resolve the same (empty) key, they are guaranteed consistent for the unbound case.
- Deleted the now-dead `ComputeAverageDuration` helper (it had a single call site — the removed branch).
- Added `BlendTreeTest.Simple1D_UnboundParam_DurationMatchesEvaluatedPose`: children with durations 1s/2s/3s and no bound param → asserts the evaluated pose is the first child (`x = 10`) **and** the duration is `1.0s`, not the old `2.0s` average. Genuine regression guard (fails under the old code).

The 2D path was checked for the same footgun — `Evaluate2D` and `Compute2DDuration` both read `GetFloat(X)`/`GetFloat(Y)` unconditionally, so they are already consistent; only the Simple1D special case diverged.

## Test plan

- `cmake --build build --target OloEngine-Tests --config Debug --parallel` → exit 0.
- `OloEngine-Tests.exe --gtest_filter=*BlendTree*` → 11/11 pass (incl. the new test).
- `OloEngine-Tests.exe --gtest_filter=*Anim*:*Blend*` → 111/111 pass, no regression.

Closes #410

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Fixed 1D blend tree duration calculations so they now match the evaluated pose when the blend parameter is left unbound.
  * Unbound blend parameters now consistently use the same default behavior for both playback evaluation and duration reporting.
  * Added coverage for the unbound-parameter case to help prevent regressions.

### CODE REVIEW

**Comment by drsnuggles8** [comment]
## 🤖 Self-review (finish-pr) @ `8d9538df0defb8a18334789aeb4c4321d88210ce`

Reviewed the full PR diff vs `master` at **medium** effort (independent correctness + conventions/cleanup finder passes).

- **Findings:** 2 · both fixed.
  - **(from @coderabbitai) Functional correctness — duration/pose could still diverge:** `Compute1DDuration` bracketed across *all* children while `Evaluate1D` filtered to *playable* ones (valid clip + positive speed), so a non-playable child between two playable ones made duration blend a different clip pair than the rendered pose — the same divergence class as #410. Fixed.
  - **Altitude — the consistency invariant was hand-maintained in two places:** rather than mirror the selection logic in both functions (fragile: a future edit to one path silently reintroduces the bug), extracted the playable filtering + threshold clamp + neighbor bracketing into a single `SelectBlend1DNeighbors` helper that both `Evaluate1D` (pose) and `Compute1DDuration` (duration) consume. Same pair + weight by construction.
- **Dismissed:** none.
- **Tests:** 2 new regression tests (unbound param; non-playable middle child) pin GetDuration == the duration of the clips Evaluate actually blends. Full BlendTree + Animation suite green locally (66/66, incl. all `Evaluate1D` branches).

File: OloEngine/src/OloEngine/Animation/BlendTree.cpp [hunk 1]
```diff
@@ -125,13 +109,13 @@ namespace OloEngine
...
+        // 1D: weighted average between the neighbors bracketing the blend param.
+        // An unbound (empty) BlendParameterX resolves to GetFloat("") == 0.0f, the
+        // same value Evaluate() feeds Evaluate1D -- so duration and pose stay
+        // consistent for an unconfigured tree (issue #410).
         if (Type == BlendType::Simple1D)
         {
-            return BlendParameterX.empty()
-                       ? ComputeAverageDuration(Children)
-                       : Compute1DDuration(Children, params.GetFloat(BlendParameterX));
+            return Compute1DDuration(Children, params.GetFloat(BlendParameterX));
```
[the remark this answers is not in the document]
Comment (drsnuggles8) [reply OloEngine/src/OloEngine/Animation/BlendTree.cpp:191]: Good catch — confirmed and fixed in 8d9538d. Compute1DDuration did bracket across all children while Evaluate1D filtered to playable ones, so a non-playable child between two playable ones diverged. Rather than mirror the filtering in both functions, I extracted the playable filtering + threshold clamp + neighbor bracketing into a shared `SelectBlend1DNeighbors` helper that both Evaluate1D (pose) and Compute1DDuration (duration) consume, so they select the same pair and weight by construction. Added a regression test (`Simple1D_NonPlayableMiddleChild_DurationMatchesEvaluatedPose`) covering exactly this case.
